blackboxopt/visualizations/visualizer.py

```diff
@@ -127,42 +128,56 @@ def create_hover_information(sections: dict) -> Tuple[str, List]:
     return template, data_columns
 
 
-def multi_objective_visualization(evaluations: List[Evaluation]):
-    if not evaluations:
-        raise NoSuccessfulEvaluationsError
-
-    # Prepare dataframe for visualization
-    df = evaluations_to_df(evaluations)
+def _prepare_for_multi_objective_visualization(
+    df: pd.DataFrame, objectives: Sequence[Objective]
+) -> Tuple[pd.DataFrame, Dict[str, List[str]]]:
+    loss_matrix = np.array(
+        [
+            get_loss_vector(known_objectives=objectives, reported_objectives=ovs)
+            for ovs in df["objectives"].to_dict("records")
+        ]
+    )
     df["pareto efficient", "pareto efficient"] = utils.mask_pareto_efficient(
-        df["objectives"].values
+        loss_matrix
     )
 
-    # Objectives will be used as dimensions in the plot
-    objective_columns = list(df["objectives"].columns)
-
     # Map column names (from field index) with "sections" shown in the mouse-over text
     # TODO: Handle case, where e.g. a column in user_info produces naming collision with
     #       a column from another group-index
     hover_sections: Dict[str, list] = {
-        "info": objective_columns + ["pareto efficient", "created_unixtime", "duration"]
+        "info": [o.name for o in objectives]
+        + ["pareto efficient", "created_unixtime", "duration"]
     }
     for section in ["configuration", "user_info", "settings", "optimizer_info"]:
         if section in df.columns:
             hover_sections[section] = list(df[section].columns)
 
-    # Create hover template and list of corresponding dataframe column names
-    hover_template, hover_data_columns = create_hover_information(hover_sections)
-
     # Drop group index for visualizing in plotly
     df.columns = df.columns.droplevel("group")
 
     # Formatting for nicer output
     df["duration"] = df["duration"].round("s").astype("str")
     df["created_unixtime"] = df["created_unixtime"].dt.round("1s")
 
+    return df, hover_sections
+
+
+def multi_objective_visualization(
+    evaluations: List[Evaluation], objectives: Sequence[Objective]
+):
```

Comment:
I :heart: it, that we can now plot a subset of objectives, too. 

Just too bad, that `Evaluation` doesn't carry the objective's direction, otherwise we could have avoid the breaking change.